src/libpriv/rpmostree-core.cxx

```diff
@@ -1765,9 +1764,19 @@ rpmostree_context_prepare (RpmOstreeContext *self, GCancellable *cancellable, GE
   /* setup sack if not yet set up */
   if (dnf_context_get_sack (dnfctx) == NULL)
     {
+      auto flags = (DnfContextSetupSackFlags) (DNF_CONTEXT_SETUP_SACK_FLAG_LOAD_UPDATEINFO | DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_FILELISTS);
```

Comment:
The following is the trail of nested functions affected by setting the `DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_FILELISTS` flag:

- rpmostree_context_prepare()
    - install_packages()
        - rpmostree_compose_builtin_install()
        - rpmostree_compose_builtin_tree()
    - rpmostree_compose_builtin_extensions()
    - prep_local_assembly()
        - rpmostree_sysroot_upgrader_prep_layering()
            - rpmostree_sysroot_upgrader_deploy()
            - deploy_transaction_execute()
                - deploy_transaction_class_init()
    - rpmostree_container_rebuild()
        - rpmostree_ex_builtin_rebuild()
            - rpmostree_builtin_install()
            - rpmostree_builtin_uninstall()
 
The majority of these functions seem to pertain to the install, upgrade, and uninstall of packages. Therefore, I believe it makes sense to check whether the installation is a filename path input (ie. `rpm-ostree install /usr/bin/htop` or `rpm-ostree install /etc/rsyncd.conf`. I'm also pretty sure you would never specify a filename path when uninstalling or upgrading, so I don't think their original behaviour would be affected by this change.

The only function I don't really understand is `deploy_transaction_class_init()`, a child function of `prep_local_assembly`. I couldn't seem to find where it's called in the code and, thus, I don't know if this change would negatively impact existing functionality. @jlebon @cgwalters Input would be greatly appreciated.